libs/core/src/lib/tabs/tab-list.component.ts

```diff
@@ -179,17 +246,147 @@ export class TabListComponent implements AfterViewInit, OnChanges, OnDestroy {
+            this._cacheTabsDimensions(this.tabItems.toArray());
+            this._listenOnPanelTabsChangeAndCollapse();
+            this._listenOnResizeAndHideItems();
+            this._hideOverflowingItems();
+        } else {
+            this._isCollapsed = false;
+        }
+    }
+
+    /** @hidden Caches width dimension of the elements in the tabs header */
+    private _cacheTabsDimensions(tabItems: TabItemDirective[]): void {
+        this._overflowTriggerWidth = Math.ceil(getElementWidth(this.overflowTrigger));
+        this._cacheTabsWidth(tabItems);
+    }
+
+    /** @hidden Caches the width of the tabs */
+    private _cacheTabsWidth(tabItems: TabItemDirective[]): void {
+        this._tabsWidth = tabItems.map(item => [item, Math.ceil(getElementWidth(item.elementRef(), true))]);
+    }
+
+    /** @hidden Open tab */
+    private _openTab(index: number): void {
+        if (this.stackContent) {
+            this._scrollToPanel(index);
+        } else {
+            this.panelTabs.forEach((tab, i) => tab.triggerExpandedPanel(i === index));
+        }
+    }
+
+    /** @hidden Setup mechanisms required for handling the stacked content behavior */
+    private _setupStackedContent(): void {
+        if (this.stackContent) {
+            timer(10)
+                .subscribe(() => this.panelTabs.forEach(tab => tab.triggerExpandedPanel(true)))
+        }
+    }
+
+    /** @hidden */
+    private _scrollToPanel(index: number): void {
+        const containerElement = this.contentContainer.nativeElement;
+        const panelElement = this._panelTabsArray[index].elementRef.nativeElement;
+        const distanceToScroll = panelElement.offsetTop - containerElement.offsetTop;
+        const maximumScrollTop = containerElement.scrollHeight - containerElement.clientHeight;
+        const currentScrollPosition = Math.ceil(containerElement.scrollTop);
+
+        if (!(currentScrollPosition === maximumScrollTop && distanceToScroll > maximumScrollTop)) {
+            this._disableScrollSpy = true;
+            fromEvent(containerElement, 'scroll')
+                .pipe(
+                    takeUntil(this._onDestroy$),
+                    debounceTime(100),
+                    first()
+                ).subscribe(() => this._disableScrollSpy = false);
+
+            containerElement.scrollTo({ top: distanceToScroll, behavior: 'smooth' });
+        }
+    }
+
+    private _listenOnPanelTabsChange(): void {
+        this.panelTabs.changes.pipe(
+            startWith(this.panelTabs),
+            takeUntil(this._onDestroy$),
+            map(panelTabs => panelTabs.toArray())
+        ).subscribe(panelTabs => {
+            this._panelTabsArray = panelTabs;
+            this._tabs = { visible: panelTabs, overflowing: [] };
+        })
+    }
+
+    private _listenOnPanelTabsChangeAndCollapse(): void {
+        const $tabItemsSource = this.tabItems.changes.pipe(
+            map(tabItems => tabItems.toArray()),
+            first()
+        );
```

Comment:
Is there any reason to have this observable?

Reply:
Yes, there is a following flow in the component:
`TabPanel(ContentProjected)` -> `TabItem`

So tab headers are build based on the tab panels. This means that when the tab panels change, tab headers are still not updated. That's why on tab panels change I subscribe to tab headers change and run collapse logic.